trustmanager/filestore.go

```diff
@@ -97,6 +150,52 @@ func (f *fileStore) GetData(name string) ([]byte, error) {
 	return data, nil
 }
 
+// GetEncrypted decrypts and returns data given a file name
+func (f *fileStore) GetEncrypted(name string, passphrase string) ([]byte, error) {
+	filePath := f.genEncryptedFilePath(name)
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the data doesn't have at least this size, it doesn't have any data.
+	if len(data) <= SaltSize+24+4 {
+		return nil, fmt.Errorf("Error while decrypting, not enough data in: %s", filePath)
+	}
+
+	// Get the salt from the first SaltSize bytes in data
+	salt := make([]byte, SaltSize)
```

Comment:
just use slices

```go
salt := data[:SaltSize]
data = data[SaltSize:]
```

Can do the same for the nonce at 187:188, assuming the above change is made it becomes:

```go
nonce := data[:gcm.NonceSize()]
data = data[gcm.NonceSize():]
```

Then `gcm.Open` just takes `data` with no additional slicing. You can ignore the reassignments to data but should definitely just make use of slicing to assign salt and nonce.